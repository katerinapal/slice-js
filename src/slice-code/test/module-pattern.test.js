import {snapSlice} from './helpers/utils'

const tests = [
  ({default: mod}) => [mod.foo()],
  ({default: mod}) => [mod.foo(), mod.baz(false)],
  ({default: mod}) => [mod.foo(), mod.baz(true)],
  ({default: mod}) => [mod.baz(true)],
  ({default: mod}) => [mod.baz(false)],
  ({default: mod}) => [mod.baz(false)],
  ({default: mod}) => [mod.foobar()],
  ({default: mod}) => [mod.foobar(), mod.baz(false)],
  ({default: mod}) => [mod.foobar(), mod.foo()],
]

tests.forEach(tester => {
  test(
    tester.toString(),
    snapSlice(require.resolve('./fixtures/module-pattern'), tester)
  )
})
