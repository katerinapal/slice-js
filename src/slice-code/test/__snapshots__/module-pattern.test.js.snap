exports[`test ({ default: mod }) => [mod.baz(false)] 1`] = `
"export default (() => {
  const mod = {};

  mod.bar = \'bar\';

  mod.baz = returnBaz => \'buzz\';

  mod.mokeypatchFn = /*slice-js-coverage-ignore ignore next*/function () {};
  mod.mokeypatchFn.favorite = \'my favorite animal is a 🐒\';

  mod.monkeypatchArrow = /*slice-js-coverage-ignore ignore next*/() => {};
  mod.monkeypatchArrow.favorite = \'my favorite animal is a 🐵\';

  return mod;
})();"
`;

exports[`test ({ default: mod }) => [mod.baz(false)] 2`] = `
"export default (() => {
  const mod = {};

  mod.bar = \'bar\';

  mod.baz = returnBaz => \'buzz\';

  mod.mokeypatchFn = /*slice-js-coverage-ignore ignore next*/function () {};
  mod.mokeypatchFn.favorite = \'my favorite animal is a 🐒\';

  mod.monkeypatchArrow = /*slice-js-coverage-ignore ignore next*/() => {};
  mod.monkeypatchArrow.favorite = \'my favorite animal is a 🐵\';

  return mod;
})();"
`;

exports[`test ({ default: mod }) => [mod.baz(true)] 1`] = `
"export default (() => {
  const mod = {};

  mod.bar = \'bar\';

  mod.baz = returnBaz => \'baz\';

  mod.mokeypatchFn = /*slice-js-coverage-ignore ignore next*/function () {};
  mod.mokeypatchFn.favorite = \'my favorite animal is a 🐒\';

  mod.monkeypatchArrow = /*slice-js-coverage-ignore ignore next*/() => {};
  mod.monkeypatchArrow.favorite = \'my favorite animal is a 🐵\';

  return mod;
})();"
`;

exports[`test ({ default: mod }) => [mod.foo(), mod.baz(false)] 1`] = `
"export default (() => {
  const mod = {};

  mod.foo = () => \'foo\';

  mod.bar = \'bar\';

  mod.baz = returnBaz => \'buzz\';

  mod.mokeypatchFn = /*slice-js-coverage-ignore ignore next*/function () {};
  mod.mokeypatchFn.favorite = \'my favorite animal is a 🐒\';

  mod.monkeypatchArrow = /*slice-js-coverage-ignore ignore next*/() => {};
  mod.monkeypatchArrow.favorite = \'my favorite animal is a 🐵\';

  return mod;
})();"
`;

exports[`test ({ default: mod }) => [mod.foo(), mod.baz(true)] 1`] = `
"export default (() => {
  const mod = {};

  mod.foo = () => \'foo\';

  mod.bar = \'bar\';

  mod.baz = returnBaz => \'baz\';

  mod.mokeypatchFn = /*slice-js-coverage-ignore ignore next*/function () {};
  mod.mokeypatchFn.favorite = \'my favorite animal is a 🐒\';

  mod.monkeypatchArrow = /*slice-js-coverage-ignore ignore next*/() => {};
  mod.monkeypatchArrow.favorite = \'my favorite animal is a 🐵\';

  return mod;
})();"
`;

exports[`test ({ default: mod }) => [mod.foo()] 1`] = `
"export default (() => {
  const mod = {};

  mod.foo = () => \'foo\';

  mod.bar = \'bar\';

  mod.mokeypatchFn = /*slice-js-coverage-ignore ignore next*/function () {};
  mod.mokeypatchFn.favorite = \'my favorite animal is a 🐒\';

  mod.monkeypatchArrow = /*slice-js-coverage-ignore ignore next*/() => {};
  mod.monkeypatchArrow.favorite = \'my favorite animal is a 🐵\';

  return mod;
})();"
`;

exports[`test ({ default: mod }) => [mod.foobar(), mod.baz(false)] 1`] = `
"export default (() => {
  const mod = {};

  mod.bar = \'bar\';

  mod.baz = returnBaz => \'buzz\';

  mod.foobar = function () {
    return \'function assignment\';
  };

  mod.mokeypatchFn = /*slice-js-coverage-ignore ignore next*/function () {};
  mod.mokeypatchFn.favorite = \'my favorite animal is a 🐒\';

  mod.monkeypatchArrow = /*slice-js-coverage-ignore ignore next*/() => {};
  mod.monkeypatchArrow.favorite = \'my favorite animal is a 🐵\';

  return mod;
})();"
`;

exports[`test ({ default: mod }) => [mod.foobar(), mod.foo()] 1`] = `
"export default (() => {
  const mod = {};

  mod.foo = () => \'foo\';

  mod.bar = \'bar\';

  mod.foobar = function () {
    return \'function assignment\';
  };

  mod.mokeypatchFn = /*slice-js-coverage-ignore ignore next*/function () {};
  mod.mokeypatchFn.favorite = \'my favorite animal is a 🐒\';

  mod.monkeypatchArrow = /*slice-js-coverage-ignore ignore next*/() => {};
  mod.monkeypatchArrow.favorite = \'my favorite animal is a 🐵\';

  return mod;
})();"
`;

exports[`test ({ default: mod }) => [mod.foobar()] 1`] = `
"export default (() => {
  const mod = {};

  mod.bar = \'bar\';

  mod.foobar = function () {
    return \'function assignment\';
  };

  mod.mokeypatchFn = /*slice-js-coverage-ignore ignore next*/function () {};
  mod.mokeypatchFn.favorite = \'my favorite animal is a 🐒\';

  mod.monkeypatchArrow = /*slice-js-coverage-ignore ignore next*/() => {};
  mod.monkeypatchArrow.favorite = \'my favorite animal is a 🐵\';

  return mod;
})();"
`;
